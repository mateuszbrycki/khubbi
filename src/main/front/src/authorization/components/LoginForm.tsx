import React from 'react'
import {useState} from 'react';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';

export interface LoginFormProps {
    readonly user: {} | null;
}

export interface LoginFormActionProps {
    readonly loginUser: (email: string, password: string) => void
}

const RegisterForm: React.FC<LoginFormProps & LoginFormActionProps> = (props) => {

    const {user, loginUser} = props

    const [email, setEmail] = useState<string>("")
    const [password, setPassword] = useState<string>("")

    return <>
        <h2>Login</h2>
        <Form onSubmit={event => {
            loginUser(email, password);
            event.preventDefault();
        }}>
            <Form.Group className="mb-3" controlId="loginFormEmail">
                <Form.Label>Email address</Form.Label>
                <Form.Control type="text" placeholder="Enter email"
                              onChange={(event) => setEmail(event.target.value)}/>
            </Form.Group>

            <Form.Group className="mb-3" controlId="loginFormPassword">
                <Form.Label>Password</Form.Label>
                <Form.Control type="password" placeholder="Password"
                              onChange={(event) => setPassword(event.target.value)}/>
            </Form.Group>
            <Button variant="primary" type="submit">
                Submit
            </Button>
        </Form>
    </>
}

export default RegisterForm