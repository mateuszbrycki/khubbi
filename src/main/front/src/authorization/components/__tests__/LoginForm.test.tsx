import renderer from "react-test-renderer";
import LoginForm from "../LoginForm";

it('renders correctly', () => {
    const tree = renderer
        .create(<LoginForm user={null} loginUser={(email, password) => null}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});