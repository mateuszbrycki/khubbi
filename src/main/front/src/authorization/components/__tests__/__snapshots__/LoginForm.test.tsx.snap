// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <h2>
    Login
  </h2>,
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="loginFormEmail"
      >
        Email address
      </label>
      <input
        className="form-control"
        id="loginFormEmail"
        onChange={[Function]}
        placeholder="Enter email"
        type="text"
      />
    </div>
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="loginFormPassword"
      >
        Password
      </label>
      <input
        className="form-control"
        id="loginFormPassword"
        onChange={[Function]}
        placeholder="Password"
        type="password"
      />
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </form>,
]
`;
